tests/test_ensemble_tools.py

```diff
@@ -246,14 +247,77 @@ def test_create_dict_sweep(
         for i in range(len(parameter_dict)):
             parameter = list(parameter_dict.keys())[i]
             scan_range = parameter_dict[parameter]
-            para_value = (
-                scan_range[0]
-                if n_divs == 1
-                else scan_range[0]
-                + (indices[i] / (n_divs - 1)) * (scan_range[1] - scan_range[0])
+            para_value = calculate_parameter_value(
+                n_dirs, scan_range[0], scan_range[1], indices[i]
             )
             n_equal_in_value, n_different_in_value = _check_parameter_in_conditions(
                 this_dir / "conditions.xml", parameter, para_value
             )
             assert n_equal_in_value == 1
             assert n_different_in_value == 0
+
+
+@pytest.mark.parametrize("n_dirs", [1, 3, 100])
+@pytest.mark.parametrize(
+    "scan_range",
+    [
+        [1.0, 3.0],
+        [10000, 20000],
+        [5.0, 4.0],
+    ],
+)
+def test_calculate_parameter_value(n_dirs: int, scan_range: list[float]) -> None:
+    """Tests the calculate_parameter_value method of ensemble_tools."""
+    parameter_values = [
+        calculate_parameter_value(n_dirs, scan_range[0], scan_range[1], i)
+        for i in range(n_dirs)
+    ]
+    # Check the returned list has the correct number of entries
+    assert len(parameter_values) == n_dirs
+    # Check the values in the generated list are unique
+    n_unique_entries = len(set(parameter_values))
+    assert len(parameter_values) == n_unique_entries
+    for value in parameter_values:
+        assert min(scan_range) <= value <= max(scan_range)
+
+
+@pytest.mark.parametrize("n_dirs", [1, 3, 100])
+@pytest.mark.parametrize(
+    "parameter_list",
+    [
+        {"particle_initial_velocity": [], "particle_charge_density": []},
+        {
+            "particle_initial_velocity": [],
+            "particle_charge_density": [],
+            "particle_number_density": [],
+        },
+    ],
+)
```

Comment:
If `return_directory_name` is changed to accept an iterable of names instead of a dictionary then the dummy values could be removed here and just lists / tuples of parameter names used.